src/snowcli/cli/nativeapp/manager.py

```diff
@@ -270,20 +277,33 @@ def _apply_package_scripts(self) -> None:
                 raise InvalidPackageScriptError(relpath, e)
 
         # once we're sure all the templates expanded correctly, execute all of them
-        if self.warehouse:
-            self._execute_query(f"use warehouse {self.warehouse}")
-
-        for i, queries in enumerate(queued_queries):
-            log.info(f"Applying package script: {self.package_scripts[i]}")
-            self._execute_queries(queries)
+        if self.package_warehouse:
+            self._execute_query(f"use warehouse {self.package_warehouse}")
+
+        try:
+            for i, queries in enumerate(queued_queries):
+                log.info(f"Applying package script: {self.package_scripts[i]}")
+                self._execute_queries(queries)
+        except ProgrammingError as err:
+            if str(err.errno) == "000606" or err.msg.__contains__(
+                "No active warehouse selected in the current session"
+            ):
+                raise MissingWarehouseError(
```

Comment:
Can we interpolate the original error number + message into this error's message? Want to make sure we don't obscure any extra information the backend might send.